src/game.js

```diff
@@ -0,0 +1,98 @@
+'use strict';
+
+const STRIKE = 10,    SPARE = 10,   ALL_PINS = 10,    MAX_ROLLS = 3,
+      MAX_FRAMES = 10;
+var   roll1 = 0,      roll2 = 1;
+
+
+var Game = function() {
+  this.frameCount = 0;
+  this.frames = [];
+  this.roll;
+  this.remainingPins;
+};
+
+Game.prototype = {
+
+  addBonuses() {
+    var [frames, l] = [this.frames, this.frames.length];
+    var [antepenult, penult, ult] = [frames[l - 3], frames[l - 2], frames[l -1]];
+    if (l > 2 && this.isStrike(antepenult) && !this.hasBonus(antepenult)) {
+      antepenult.push(ult[roll1]);
+      ult[roll2] ? penult.push(ult[roll1], ult[roll2]) : penult.push(ult[roll1]);
+    } else if (l > 1 && this.isStrike(penult) && !this.hasBonus(penult)) {
+      ult[roll2] && ult[roll2] !== 10 ?
+      penult.push(ult[roll1], ult[roll2]) : penult.push(ult[roll1]);
+    } else if (l > 1 && this.isSpare(penult) && !this.hasBonus(penult)) {
+      penult.push(ult[roll1]);
+    }
+    return this.frames;
+  },
+
+  flattenAndSum(array) {
+    array = array.reduce(function(acc, curr) {
+      return acc.concat(curr);
+    }, []);
+
+    return array = array.reduce(function (acc, curr) {
+      return acc + curr;
+    });
+  },
+
+  addRollToFrame() {
+    if (this.frames.length < MAX_FRAMES && !this.frames[this.frameCount]) {
+      this.frames[this.frameCount] = [this.roll];
+    } else if (this.frames.length < MAX_FRAMES) {
+      this.frames[this.frameCount].push(this.roll);
+    } else {
+      this.frames[this.frameCount] ?
+        this.frames[this.frameCount].push(this.roll)
+        : this.frames[this.frameCount] = [this.roll];
+    }
+    return this.frames;
+  },
+
+  checkFrameComplete() {
+    if (this.frames[this.frameCount][roll1] === STRIKE
+        || this.frames[this.frameCount].length === 2) {
+      this.addBonuses();
+      this.frameCount ++;
+    }
+  },
+
+  checkFrameTenComplete() {
+    this.addBonuses();
+    if (this.frames[this.frameCount].length === MAX_ROLLS
+      || this.frames[this.frameCount][roll1]
```

Comment:
Whenever I see a multidimensional array, I wonder whether I'm missing a class. This feels similar to reaching into the state of another object, i.e. manipulating the roll property of a frame class from the game class. Also, we're doing a lot of calculations with this.frames[this.frameCount] - that kind of repetition is another indication that this logic should be encapsulated elsewhere. 

I would expect instead to be seeing something like:
```
this.activeFrame().isFrameComplete() 
```
And move the logic into a frame class.